src/wpool_time_checker.erl

```diff
@@ -84,26 +84,37 @@ handle_call({add_handler, Handler}, _, State = #state{handlers = Handlers}) ->
 %%%===================================================================
 %% @private
 -spec handle_info(any(), state()) -> {noreply, state()}.
-handle_info({check, Pid, TaskId, Runtime}, State) ->
+handle_info({check, Pid, TaskId, Runtime, WarningsLeft}, State) ->
   case erlang:process_info(Pid, dictionary) of
     {dictionary, Values} ->
       run_task(
         TaskId, proplists:get_value(wpool_task, Values), Pid,
-        State#state.wpool, State#state.handlers, Runtime);
+        State#state.wpool, State#state.handlers, Runtime, WarningsLeft);
     _ -> ok
   end,
   {noreply, State};
 handle_info(_Info, State) -> {noreply, State}.
 
-run_task(TaskId, {TaskId, _, Task}, Pid, Pool, Handlers, Runtime) ->
-  Args = [ {alert, overrun}, {pool, Pool}, {worker, Pid}, {task, Task}
-         , {runtime, Runtime}],
-  _ = [catch Mod:Fun(Args) || {Mod, Fun} <- Handlers],
+run_task(TaskId, {TaskId, _, Task}, Pid, Pool, Handlers, Runtime, 1) ->
+  send_reports(Handlers, Pool, Pid, Task, Runtime),
```

Comment:
It would be nice to add some indication to the report args noting that the process is killed this time, maybe instead of `{alert, overrun}` we can use `{alert, max_overrun_limit}` or something like that to indicate that the worker is killed this time.

Reply:
I agree